--- src/prepare.c
+++ src/prepare.c
@@ -249,6 +249,10 @@
*/
for(i=0; i<ArraySize(meta); i++)
pDb->pSchema->schema_cookie = meta[BTREE_SCHEMA_VERSION-1];
@@ -483,7 +487,10 @@
** set Parse.rc to SQLITE_SCHEMA. */
sqlite3BtreeGetMeta(pBt, BTREE_SCHEMA_VERSION, (u32 *)&cookie);
assert( sqlite3SchemaMutexHeld(db, iDb, 0) );
-    if( cookie!=db->aDb[iDb].pSchema->schema_cookie )else{
iGen = iMeta = 0;
@@ -3155,6 +3159,8 @@
assert( DbMaskTest(p->btreeMask, iDb) );
sqlite3BtreeGetMeta(db->aDb[iDb].pBt, iCookie, (u32 *)&iMeta);
+  if ( rc == SQLITE_OK && db->errCode == SQLITE_BUSY )
+    rc = db->errCode;
pOut = out2Prerelease(p, pOp);
pOut->u.i = iMeta;
break;
