--- src/os_common.h
+++ src/os_common.h
@@ -58,13 +58,13 @@
** is used for testing the I/O recovery logic.
*/
-int sqlite3_io_error_hit = 0;
-int sqlite3_io_error_hardhit = 0;
-int sqlite3_io_error_pending = 0;
-int sqlite3_io_error_persist = 0;
-int sqlite3_io_error_benign = 0;
-int sqlite3_diskfull_pending = 0;
-int sqlite3_diskfull = 0;
+extern int sqlite3_io_error_hit;
+extern int sqlite3_io_error_hardhit;
+extern int sqlite3_io_error_pending;
+extern int sqlite3_io_error_persist;
+extern int sqlite3_io_error_benign;
+extern int sqlite3_diskfull_pending;
+extern int sqlite3_diskfull;
#define SimulateIOErrorBenign(X) sqlite3_io_error_benign=(X)
#define SimulateIOError(CODE)  \
if( (sqlite3_io_error_persist && sqlite3_io_error_hit) \
@@ -96,7 +96,7 @@
** When testing, keep a count of the number of open files.
*/
-int sqlite3_open_file_count = 0;
+extern int sqlite3_open_file_count;
#define OpenCounter(X)  sqlite3_open_file_count+=(X)
#define OpenCounter(X)
--- src/os_win.c
+++ src/os_win.c
@@ -20,6 +20,17 @@
*/
+
+int sqlite3_io_error_hit = 0;
+int sqlite3_io_error_hardhit = 0;
+int sqlite3_io_error_pending = 0;
+int sqlite3_io_error_persist = 0;
+int sqlite3_io_error_benign = 0;
+int sqlite3_diskfull_pending = 0;
+int sqlite3_diskfull = 0;
+int sqlite3_open_file_count = 0;
+
+
--- src/os_unix.c
+++ src/os_unix.c
@@ -266,6 +266,17 @@
*/
+
+int sqlite3_io_error_hit = 0;
+int sqlite3_io_error_hardhit = 0;
+int sqlite3_io_error_pending = 0;
+int sqlite3_io_error_persist = 0;
+int sqlite3_io_error_benign = 0;
+int sqlite3_diskfull_pending = 0;
+int sqlite3_diskfull = 0;
+int sqlite3_open_file_count = 0;
+
+
